app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,85 @@
+#TASKS = ["Laundry", "Feed Babette", "Clean room", "Sleep"] 
+class TasksController < ApplicationController
+  
+  def index
+    @tasks = Task.all
+  end
+  
+  def show
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: params[:id])
+    if @task.nil?
+      redirect_to root_path
+      return
+    end
+  end
+  
+  def new
+    @task = Task.new
+  end
+  
+  def create
+    @task = Task.new(name: params[:task][:name], description: params[:task][:description], completed: params[:task][:completed])
+    if @task.save
+      redirect_to task_path(@task.id)
+    else
+      redirect_to root_path
+    end
+  end 
+  
+  def edit
+    task_id = params[:id].to_i
+    @task = Task.find_by(id: params[:id] )
```

Comment:
Same as my comments above in the `show` action, consider refactoring these two lines